environment.sh

```diff
@@ -36,52 +38,95 @@ done
 # Parse environment name
 env_name=$(basename "`pwd`")
 env_name+="_$env_type"
+branch_name=$(git rev-parse --abbrev-ref HEAD)
+# Determine which requirements.txt to install from
+if [ $env_type == 'simulation' ]; then
+  install_file="requirements.txt"
+elif [ $env_type == 'artifact' ]; then
+  install_file="artifact_requirements.txt"
+else
+  echo "Invalid environment type. Valid argument types are simulation and artifact."
+  return 
+fi
+requirements_modification_time="$(date -r $install_file '+%Y-%m-%d %H:%M:%S')"
+
+# Pull repo to get latest changes from remote if remote exists
+git ls-remote --exit-code --heads origin $branch_name >/dev/null 2>&1
+exit_code=$?
+if [[ $exit_code == '0' ]]; then
+  git fetch --all
+  echo "Git branch '$branch_name' exists in the remote repository"
```

Comment:
Maybe `echo "Git branch '$branch_name' exists in the remote repository; pulling latest changes"`